### PR TITLE
Added --ver-dep flag to sui cli

### PR DESCRIPTION
**Pull request by ronny-mysten**
Also updated content in section to comply with style guide.

### CODE REVIEW

File: doc/src/build/cli-client.md [hunk 1]
```diff
@@ -507,25 +507,24 @@ id: 0x471c8e241d0473c34753461529b70f9c4ed3151b[1]
...
 ## Publish packages
 
-In order for user-written code to be available in Sui, it must be
-*published* to Sui's [distributed ledger](../learn/how-sui-works.md#architecture).
-Please see the [Move developer documentation](move/index.md) for a
-description on how to [write a simple Move code package](move/write-package.md),
-which we can publish using Sui client's `publish` command.
+You must publish packages to the Sui [distributed ledger](../learn/how-sui-works.md#architecture) for the code you developed to be available in Sui. To publish packages with the Sui client, use the `publish` command.
+
+The publish command requires that you specify the directory where your package lives using the `--path` parameter. The value is the path to the `my_move_package` as per the [package creation description](move/write-package.md). You must also provide a `gas` object to pay for publishing the package and a `gas-budget` value.
```
Comment (amnn) [inline doc/src/build/cli-client.md:512]: As of 7c68497 the `path` parameter is no longer passed as a flag (`--path`) it's an optional positional parameter the defaults to the current working directory, so we might as well fix the docs up to match that in this pass as well!

Also, nit: Is the "the" necessary here?

```suggestion
The publish command requires that you specify the directory where your package lives using the `--path` parameter. The value is the path to `my_move_package` as per the [package creation description](move/write-package.md). You must also provide a `gas` object to pay for publishing the package and a `gas-budget` value.
```

File: doc/src/build/cli-client.md [hunk 2]
```diff
@@ -507,25 +507,24 @@ id: 0x471c8e241d0473c34753461529b70f9c4ed3151b[1]
...
 
-> **Important:** All calls to functions in the `debug` module must be removed from no-test code
-> before the new module can be published (test code is marked with the `#[test]` annotation).
+Parameters used for the `publish` command example:
+* `--path` - Defines the path to the Move package to publish.
+* `--gas` - The Coin object used to pay for gas.
+* `--gas-budget` - Gas budget for running module initializers. 
+* `--verify-dependencies` - Optional flag to have the CLI check that a dependency exists on-chain.
```
Comment (amnn) [inline doc/src/build/cli-client.md:518]: Suggestion to indicate that it checks all dependencies, and beyond just existing, they need to match (please feel free to rephrase).

```suggestion
* `--verify-dependencies` - Optional flag to have the CLI check that all dependencies match their on-chain counterparts. 
```

File: doc/src/build/cli-client.md [hunk 3]
```diff
@@ -549,10 +548,22 @@ that the location of the package's sources is in the `PATH_TO_PACKAGE`
 environment variable):
 
 ```shell
-$ sui client publish --path $PATH_TO_PACKAGE/my_move_package --gas-budget 30000
+$ sui client publish --path $PATH_TO_PACKAGE/my_move_package --gas 0xc8add7b4073900ffb0a8b4fe7d70a7db454c2e19 --gas-budget 30000 --verify-dependencies
 ```
 
-The result of running this command should look as follows:
+The call uses the optional `--verify-dependencies` flag to verify the bytecode for dependencies found at their respective published addresses matches the bytecode you get when compiling that dependency from source code. If the bytecode for a dependency does not match, your package does not publish and you receive an error message indicating which package and module the mismatch was found:
```
Comment (amnn) [inline doc/src/build/cli-client.md:554]:
```suggestion
The call uses the optional `--verify-dependencies` flag to verify the bytecode for dependencies found at their respective published addresses matches the bytecode you get when compiling that dependency from source code. If the bytecode for a dependency does not match, your package does not publish and you receive an error message indicating at which package and module the mismatch was found:
```

**Review by amnn — CHANGES_REQUESTED** [review]
Overall looks good!  I noticed that there's another change in the `publish` CLI format that we should incorporate in the docs change, so requesting changes for that.

File: doc/src/build/cli-client.md (hunk 1 above)
Comment (ronny-mysten) [reply doc/src/build/cli-client.md:512]: It is not.

**Review by randall-Mysten — APPROVED** [review]

**Pull request merged**